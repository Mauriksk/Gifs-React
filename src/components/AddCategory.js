import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategorie} ) => {

    const [inputValue, setinputValue] = useState('')

    const handleInputChange = (e) => {
        setinputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setCategorie( animes => [ inputValue, ...animes ])
            setinputValue('')
        }
        
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
            type="text"
            value={ inputValue }
            onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategorie: PropTypes.func.isRequired
}


