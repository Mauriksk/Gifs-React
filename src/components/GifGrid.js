import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category} ) => {

    
    const { data:images , loading } = useFetchGifs( category );

    //Este hock me permite poner dependencias en el arreglo, si el arreglo esta vacio la funcion dentro solo se ejecutara una ves
    /* useEffect(() => {
        getGif( category )
            .then(setimages)
    }, [ category ]) */

    //Le coloco el category al array para que no joda el warnin y para que en caso de que cambie la categoria
    //vuelva a permitir ejecutar de nuevo la peticion

    

    return (
        <>

        <h3 className="animate__animated animate__pulse"> {category} </h3>
        <div className="card-grid">
        {loading && <p className="animate__animated animate__flash">Loading</p>}
            {
                images.map( (img) => (
                    <GifGridItem 
                        key={img.id}
                        //Otra forma de enviarlas props (todas)
                        { ...img }
                    />
                ))
            }
        </div> 

        </>
    )
}
