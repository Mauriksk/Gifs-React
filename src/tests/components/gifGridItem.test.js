import React from 'react';
import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe('Prueba gifGridItem', () => {

    const url = 'https://www.upwork.com/ab/find-work/recommended'
    const title = 'Hola soy un titulo de Prueba'
    const wrapper = shallow( <GifGridItem url={url} title={title} /> )

    test('Debe retornar el Componente <GifGridItem />.', () => {

        expect( wrapper ).toMatchSnapshot()

    })

    test('El parafo recibe el title', () => {
        
        const p = wrapper.find( 'p' )
        
        expect( p.text().trim() ).toBe( title )

    })

    test('Evaluacion de la imagen', () => {
        
        const img = wrapper.find( 'img' )
        //El .prop() es un metodo de enzyme 
        expect( img.prop( 'src' ) ).toBe( url )
        expect( img.prop( 'alt' ) ).toBe( title )

    })

    test('Evaluando clases', () => {
        
        const div = wrapper.find( 'div' )
        const className = div.prop( 'className' )
        expect( className.includes( 'animate__backInRight' ) ).toBe( true )

    })
    
    
    

} )