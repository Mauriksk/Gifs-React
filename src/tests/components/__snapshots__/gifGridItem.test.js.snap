// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba gifGridItem Debe retornar el Componente <GifGridItem />. 1`] = `
<div
  className="card animate__animated animate__backInRight"
>
  <img
    alt="Hola soy un titulo de Prueba"
    src="https://www.upwork.com/ab/find-work/recommended"
  />
  <p>
    Hola soy un titulo de Prueba
  </p>
</div>
`;
